source/common/router/router.cc

```diff
@@ -463,6 +463,9 @@ void Filter::setDecoderFilterCallbacks(Http::StreamDecoderFilterCallbacks& callb
 }
 
 void Filter::cleanup() {
+  if (upstream_request_) {
```

Comment:
I was a little worried about this change because I wasn't sure if there are other paths in which the upstream_request_ can be destroyed and we would lose any data that we might have otherwise had. I think this is correct from a quick look on my own, but can you add a comment above this that says that all upstream_request_ destructions (other than retry where we don't save timings to downstream) will go through this function? (Or something like that.)